resources/functionMetadata.php

```diff
@@ -1444,8 +1455,8 @@
 	'posix_uname' => ['hasSideEffects' => false],
 	'pow' => ['hasSideEffects' => false],
 	'preg_grep' => ['hasSideEffects' => false],
-	'preg_last_error' => ['hasSideEffects' => false],
-	'preg_last_error_msg' => ['hasSideEffects' => false],
+	'preg_last_error' => ['hasSideEffects' => true],
```

Comment:
https://3v4l.org/O43Rb

what side effects does it have?

Reply:
Thanks to this line we solved this issue https://github.com/phpstan/phpstan/issues/10342

Reply:
Sure, I mean we should differentiate between pure and side effects. In my understanding "pure" is about return value, "side effects" are about effects on other functions. So I would expect this function to have no side effects, but be pure instead.